### PR TITLE
[Lllama] Update tokenization code to ensure parsing of the special tokens [core]

### LINKED ISSUE #23818: LLaMATokenizerFast works abnormally
### System Info

platform==Ubuntu18.04
python==3.10
transformers==4.29.2

### Who can help?

@ArthurZucker

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

`</s>` is the special token of LLaMATokenizer(Fast), it is expected that `</s>` can be recognized as a single token when encoding the text. However, it can be shown that the two tokenizers behave differently:
```python
>>> t1 = transformers.AutoTokenizer.from_pretrained("huggyllama/llama-7b", use_fast=True)
>>> t2 = transformers.AutoTokenizer.from_pretrained("huggyllama/llama-7b", use_fast=False)
>>> text = "I love you.</s>"
>>> t1(text)
>>> {'input_ids': [1, 306, 5360, 366, 21106, 29879, 29958], 'token_type_ids': [0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1, 1, 1, 1]}
>>> t2(text)
>>> {'input_ids': [1, 306, 5360, 366, 29889, 2], 'attention_mask': [1, 1, 1, 1, 1, 1]}
```
also, LLaMATokenizerFast returns `token_type_ids` but LLaMATokenizer does not.

### Expected behavior

LLaMATokenizerFast to be consistent with LLaMATokenzier.

### PR DESCRIPTION
**Pull request by ArthurZucker**
# What does this PR do?
Adresses the issues with the fast tokenizer of LLama. Namely: 
- nit making it return token type ids. 
- the added tokens are not correctly encoded. 

There seems to be an issue with the conversion: before the python layer, just loading the tokenizer_config.json file with the rust backend still produced: `tokenizer.encode("this is not<s>").tokens`, ` ['<s>', '▁This', '▁is', '▁not', '</', 's', '>']`

### CODE REVIEW

**Comment by ArthurZucker** [comment]
Ok, narrowed it down to this line: 
```python 
        # Check all our special tokens are registered as "no split" token (we don't cut them) and are in the vocab
        added_tokens = tokenizer.sanitize_special_tokens()
```
When converting the model from a slow one, the tokenizer correctly processes the inputs up until this point. Meaning that before, the special tokens where already registered as special tokens, but adding them once more most probably breaks the internal regex. Still checking but should be this.

**Comment by ArthurZucker** [comment]
After debugging with @Narsil it seems that the special tokens have to be not normalised, otherwise the normalizer prepends a space when adding it, which is why the token is not recognized. I suspect that there is another bug, as I tried with special tokens set to normalized = True (when calling `from_slow=True`+commenting `self._sanitize_special_tokens`) but the current should fix the conversion. 

All config.json have to be update since they are probably using `normalized = True`. Also a strange thing is that this is not the case for the slow tokenizer, which means that the logic of handling normalization is different/ does not happen in the same order.

**Review by sgugger — APPROVED** [review]
Thanks for the fix!

**Comment by ArthurZucker** [comment]
Submitting PR to update the tokenizer.json file on the `huggyllama` repos
